Src/Base/AMReX_GpuAtomic.H

```diff
@@ -386,8 +387,11 @@ namespace detail {
         sycl::atomic_ref<T,mo,ms,as> a{*m};
         return a.fetch_max(value);
 #else
-        amrex::ignore_unused(m,value);
-        return T(); // should never get here, but have to return something
+        AMREX_IF_ON_DEVICE(( return atomixMax(m, value); ))
```

Comment:
```suggestion
        AMREX_IF_ON_DEVICE(( return atomicMax(m, value); ))
```

This and `Min_device` can probably be written as
```
#if defined(__SYCL_DEVICE_ONLY__)
  ...
#else
    AMREX_IF_ON_DEVICE(...);
    AMREX_IF_ON_HOST(...);
#endif
```

Reply:
and other functions like that.

Reply:
Fixed.